src/frontend/src/pages/PIAIntakeForm/index.tsx

```diff
@@ -67,14 +84,48 @@ const PIAIntakeFormPage = () => {
         setPiaModalTitleText(Messages.Modal.Save.TitleText.en);
         setPiaModalParagraph(Messages.Modal.Save.ParagraphText.en);
         break;
+      case 'edit':
+        setPiaModalConfirmLabel(Messages.Modal.Edit.ConfirmLabel.en);
+        setPiaModalCancelLabel(Messages.Modal.Edit.CancelLabel.en);
+        setPiaModalTitleText(Messages.Modal.Edit.TitleText.en);
+        setPiaModalParagraph(Messages.Modal.Edit.ParagraphText.en);
+        break;
       default:
         break;
     }
     setShowPiaModal(true);
   };
 
-  const handleModalClose = () => {
+  const handleModalClose = async (event: any) => {
     setShowPiaModal(false);
+    // call backend patch endpoint to save the change
+    event.preventDefault();
+    const requestBody: IPIAIntake = {
+      title: title,
+      ministry: ministry,
+      branch: branch,
+      drafterName: drafterName,
+      drafterEmail: drafterEmail,
+      drafterTitle: drafterTitle,
+      leadName: leadName,
+      leadEmail: leadEmail,
+      leadTitle: leadTitle,
+      mpoName: mpoName,
+      mpoEmail: mpoEmail,
+      initiativeDescription: initiativeDescription,
+      initiativeScope: initiativeScope,
+      dataElementsInvolved: dataElementsInvolved,
+      hasAddedPiToDataElements: hasAddedPiToDataElements,
+      riskMitigation: riskMitigation,
+    };
+    try {
+      await HttpRequest.patch<IPIAResult>(
+        API_ROUTES.GET_PIA_INTAKE.replace(':id', `${pia.id}`),
```

Comment:
I believe Kushal mentioned this as well, but shouldn't we have something like `API_ROUTES.UPDATE_PIA_INTAKE` which points to the update endpoint?

Reply:
yes, update to patch one